app/renderer/ui/components/list/index.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react"
+import { List } from "antd"
+
+import { ButtonOption } from "app/renderer/ui/components/button"
+
+const styles = require("./style.scss")
+
+export interface Iprops {
+  grid?: {} | undefined
+  classNameList?: string
+  classNameItem?: string
+  dataSource: Array<(string | { text: any; onClick: any } | any)>
+ // dataSource: (string | { text: any; onClick: any })[]
+  borderer?: boolean
+  renderItem?: any
+}
+
+/**
+ * List UI component
+ *
+ * @export
+ * @class List
+ * @extends {React.Component<Iprops>}
+ */
+
+export default class DefaultList extends React.Component<Iprops> {
+  // tslint:disable-next-line: completed-docs
+  public render() {
+    const ListItem = this.props.renderItem
+    const isClickable = (item: any) => item.onClick && item.text
```

Comment:
For the sake of simplicity, our `render()`s should be expressions or at least contain as few statements as possible, IMO.

For example, internal function definitions should be moved out of the component class.

As an exception, methods depending on properties of the class could be transformed into private class methods too, although that's not strictly necessary as they can receive such properties as parameters.

Think of this: right now those functions are being redefined every single time `render()` is called, which can happen quite often as a reaction to app state updates. That's a redundant CPU/memory expenditure and can have a significant negative impact on drawing times.